### PR TITLE
Friends

### PR DESCRIPTION
**Pull request by mikolajmirko**
- Adding friends
- Removing friends
- Displaying friend's plans
- New plan form mode: copy
- Dialogs with user code and copy to clipboard function
- UI changes:
  - Popup menu icons
  - Plan form loader and success snackbars
 - Code refactor

### CODE REVIEW

File: lib/pages/caregiver/forms/plan_form_page.dart [hunk 1]
```diff
@@ -55,35 +56,45 @@ class _CaregiverPlanFormPageState extends State<CaregiverPlanFormPage> {
 		formKey = GlobalKey<FormState>();
     return BlocConsumer<PlanFormCubit, PlanFormState>(
 			listener: (context, state) {
-				if (state is PlanFormSubmissionSuccess)
-					Navigator.of(context).pop(); // TODO also show some visual feedback?
-				else if (state is PlanFormDataLoadSuccess)
+				if (state is PlanFormSubmissionSuccess) {
+					if(formType == AppFormType.copy)
+						Navigator.of(context).pushNamed(AppPage.caregiverPlans.name);
```
Comment (stasgora) [inline lib/pages/caregiver/forms/plan_form_page.dart:61]: Not a good idea: what if user presses _back_? Either pop once or twice or, if you want to navigate to caregiver plans you can try: `Navigator.of(context).popUntil(ModalRoute.withName(AppPage.caregiverPanel.name));` before pushing plans. **Could be off by one, test first**

File: i18n/en_US.json [hunk 2]
```diff
@@ -184,7 +189,10 @@
 				"content": {
 					"noPlansText": "This friend didn't add any plans yet",
 					"plansTitle": "Friend's plan",
```
Comment (stasgora) [inline i18n/en_US.json:191]:
```suggestion
					"plansTitle": "Friend's plans",
```

File: lib/pages/caregiver/panel_page.dart [hunk 3]
```diff
@@ -60,14 +62,14 @@ class _CaregiverPanelPageState extends State<CaregiverPanelPage> {
 				title: '$_pageKey.content.caregiverProfilesTitle',
 				subtitle: '$_pageKey.content.caregiverProfilesSubtitle',
 				noElementsMessage: '$_pageKey.content.noCaregiverProfilesAdded',
-				headerAction: UIButton('$_pageKey.header.addCaregiver', () => {}, AppColors.caregiverButtonColor, Icons.add),
+				headerAction: UIButton('$_pageKey.header.addCaregiver', () => showAddFriendDialog(context, () => context.bloc<CaregiverPanelCubit>().doLoadData()), AppColors.caregiverButtonColor, Icons.add),
```
Comment (stasgora) [inline lib/pages/caregiver/panel_page.dart:65]: What's not working without that refresh callback? We should be able to handle it in a more elegant way

**Review by stasgora — CHANGES_REQUESTED** [review]

File: lib/pages/caregiver/panel_page.dart (hunk 3 above)
Comment (mikolajmirko) [reply lib/pages/caregiver/panel_page.dart:65]: I tried to do it with user auth state update, but it didn't work. It needs to get friend name and reload the list (only the code is provided). Probably there is a way, I just don't know how to do it :c

**Comment by mikolajmirko** [comment]
Google photoURL is not tested yet due to problems with signing in with Google.

I added our loader to loading page, it still has the same problem as it had earlier - its not loading fast enough to show itself, but it will come up if the loading page is displayed for longer than 1 second. Otherwise there is only a "loading..." text.

**Review by stasgora — APPROVED** [review]

**Pull request merged**